Messy.py

```diff
@@ -0,0 +1,252 @@
+        except IOError:
+            print(f"(assuming {remote_folder}/ already exists)")
+
+    def get_files(self, remote_folder: str, local_folder: str = "./") -> None:
+        """
+        Get files from the remote server
+        """
+        if not os.path.exists(local_folder):
+            os.mkdir(local_folder)
+
+        for entry in self.sftp_client.listdir_attr(remote_folder):
+            remotepath = remote_folder + "/" + entry.filename
+            localpath = os.path.join(local_folder, entry.filename)
+            mode = entry.st_mode
+            if S_ISDIR(mode):
+                try:
+                    os.mkdir(localpath)
+                except OSError:
+                    pass
+                self.get_files(remotepath, localpath)
+            elif S_ISREG(mode):
+                self.sftp_client.get(remotepath, localpath)
+            else:
+                # TODO: throw an exception here
+                print("Unknown type encountered when running get_files method")
+
+    def put_files(
+        self,
+        remote_folder: str = "~",
+        local_folder: str = "./",
+        verbose: bool = False,
+    ) -> None:
+        """
+        Copy files to the remote server
+        """
+        if remote_folder == "~":
+            remote_folder = os.path.expanduser("~")
+
+        local_folder_dirname = os.path.dirname(local_folder)
+        local_folder_basename = os.path.basename(local_folder)
+
+        if verbose:
+            print(f"local_folder_dirname: {local_folder_dirname}")
+            print(f"local_folder_basename: {local_folder_basename}")
+
+        for dirpath, dirnames, filenames in os.walk(local_folder):
+            file_dirpath = dirpath[len(local_folder) + 1 :]
+            remote_path = os.path.join(
+                remote_folder, local_folder_basename, file_dirpath
+            )
+            if verbose:
+                print("\n" + "-" * 20 + "\n")
+                print(f"dirpath: {dirpath}")
+                print(f"dirnames: {dirnames}")
+                print(f"remote_path: {remote_path}")
+                print(f"file_dirpath: {file_dirpath}")
+
+            try:
+                self.sftp_client.listdir(remote_path)
+            except IOError:
+                self.sftp_client.mkdir(remote_path)
+
+            for filename in filenames:
+                from_path = os.path.join(dirpath, filename)
+                to_path = os.path.join(remote_path, filename)
+                self.sftp_client.put(from_path, to_path)
+                if verbose:
+                    print(
+                        f"\ncopying: {filename}\n  from: {from_path}\n  to: {to_path}"
+                    )
+
+    def submit_job(self, job_file: str) -> None:
+        """
+        Submit job to SLURM cluster
+        """
+        stdin, stdout, stderr = self.ssh_client.exec_command(
+            f"sbatch {job_file}"
+        )
+        job_id = int(stdout.read().decode().split()[-1])
```

Comment:
Would you still have a job_id if you have stderr output?

Reply:
Added an issue about this.